src/proj_equirect.jl

```diff
@@ -47,36 +70,104 @@ end
 
 end
 
-typealias_def(::Type{F}) where {B,M<:ProjEquiRect,T,A,F<:EquiRectField{B,M,T,A}} = "EquiRect$(typealias(B)){$(typealias(A))}"
 function Base.summary(io::IO, f::EquiRectField)
     @unpack Ny,Nx,Nbatch = f
     print(io, "$(length(f))-element $Ny×$Nx$(Nbatch==1 ? "" : "(×$Nbatch)")-pixel ")
     Base.showarg(io, f, true)
 end
 
 
-### basis conversion
+# Field Basis
+# ================================================
+# NOTE: I still don't fully understand what AzFourer, Map ... etc is. 
+# EquiRectAzFourier, EquiRectQUAzFourier ... seem to be aliases for a base field type.
+# Are then AzFourier, Map just methods for making the conversion? Why not use the 
+# field types themselfs for the conversion?
```

Comment:
There's only one single concrete field type:

```julia
struct BaseField{B, M<:FieldMetadata, T, A<:AbstractArray{T}} <: Field{B, T}
    arr :: A
    metadata :: M
    function (::Type{F})(arr::A, metadata::M) where {B,M<:FieldMetadata,T,A<:AbstractArray{T},F<:BaseField{B}}
        new{B,M,T,A}(arr, metadata)
    end
end
```

which takes a type parameter `B` which is meant for any info needed at compile-time (right now, just a `Basis` type, many of them defined [here](https://github.com/marius311/CMBLensing.jl/blob/40ba38d0ecba89aca5e3c8eb7cd0266395a5cced/src/generic.jl#L40-L53), plus the special `AzFourier` and `QUAzFourier` in `proj_equirect.jl`), and the `metadata` which is for stuff needed also at run-time (which is the `Proj` object). `FlatMap`, `EquiRectAzFourier` are all just type aliases for different kinds of `BaseField`. E.g a `FlatMap` is a
```julia
julia> FlatMap
BaseField{Map, M<:CMBLensing.FlatProj, T, A<:(AbstractArray{T, N} where N)} where {M<:CMBLensing.FlatProj, T, A<:(AbstractArray{T, N} where N)}
```
These aliases are all defined by `make_field_aliases`, which is reused for `Flat` fields, `EquiRect` fields, etc...

We could certainly make `FlatMap(f)` work, but my motivation for having it be `Map(f)` is that it allows writing more generic code, which would work if `f` is a `FlatField`, `EquiRectField`, etc... 

Hope that helps clarify things a bit.

Reply:
No worries, its just my lazy brain resisting something about it. Perhaps its the ordering of `EquiRectAzFourier` ... feels like it would be a bit more copacetic if it was `AzFourierEquiRect` as short for `BaseFieldAzFourierEquiRect` as short for the type union `BaseField{AzFourier, ProjEquiRect, T, A} where {T, A}`. Lets not worry about it tho:)